velox/expression/VectorWriters.h

```diff
@@ -534,14 +536,7 @@ struct VectorWriter<Generic<T, comparable, orderable>>
       castWriter_->ensureSize(size);
     } else {
       if (vector_->size() < size) {
-        if (vector_->typeKind() == TypeKind::ROW) {
-          // Avoid calling resize on all children by adding top level nulls
-          // only.
-          vector_->asUnchecked<RowVector>()->appendNulls(
-              size - vector_->size());
-        } else {
-          vector_->resize(size, false);
-        }
+        vector_->resize(size, false);
```

Comment:
does this also require calling resize on children before this call? similar to the reason mentioned at L280?

Reply:
@laithsakka , do we cache data_ for the generic case ? I believe its not required here, correct me if not.